tilavarauspalvelu/api/graphql/types/reservation_unit/types.py

```diff
@@ -180,6 +180,9 @@ class Meta:
     @classmethod
     def filter_queryset(cls, queryset: models.QuerySet, info: GQLInfo) -> models.QuerySet:
         # Always hide archived reservation units
+        if getattr(info.return_type, "name", None) in ["ReservationNode", "ReservationNodeConnection"]:
+            # Allow fetching data for archived reservation units, through relations from ReservationNode
+            return queryset
```

Comment:
This feels a bit hacky, @matti-lamppu is there some "correct" way to do this? 🤔

Reply:
Hmm, I don't think there is really any good way to do this. The only "good" way I can think of is to add custom resolvers for the reservation unit endpoint, as well as all relations on all related models, and then you could filter out the archived reservation units where necessary. The issue there is that you might expose archived reservation units from some relation if you forget something or things gets added later.

Reply:
All right, I will keep this as is then. I don't really like the "magic strings" but we have tests in case something breaks this.